ci/docker-profile/profile_entry.sh

```
@@ -0,0 +1,223 @@
+
+# ---------------------------------------------------------------- setup
+log "relaxing perf/bpf restrictions (best effort)"
+sysctl -qw kernel.perf_event_paranoid=-1 2>/dev/null || log "WARN: could not set perf_event_paranoid"
+sysctl -qw kernel.kptr_restrict=0 2>/dev/null || true
+mount -t debugfs debugfs /sys/kernel/debug 2>/dev/null || true
+
+# The pinned toolchain download can exceed soldr's default 60s
+# no-output watchdog on first run; ensure it explicitly (same pattern
+# as ci/docker-mac-cross/build.sh) with a generous timeout.
+export SOLDR_COMMAND_OUTPUT_TIMEOUT_SECS="${SOLDR_COMMAND_OUTPUT_TIMEOUT_SECS:-1800}"
+cd /work
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)